sdk/python/kfp/v2/compiler/compiler.py

```diff
@@ -286,10 +288,25 @@ def _get_inputs_outputs(
     inputs = collections.defaultdict(set)
     outputs = collections.defaultdict(set)
 
+    # Fix possible missing type -- PipelineParam parsed from command line args
+    # doesn't contain the type information, as the `.param_type` is not included
+    # during PipelineParam serialization.
+    all_params = {param.pattern: param for param in args}
+    for op in pipeline.ops.values():
+      for param in op.inputs + list(op.outputs.values()) + list(
+          condition_params[op.name]):
+        if param.pattern not in all_params:
```

Comment:
The `param.pattern` is a bit hacky and fragile.
Can we detect the situation in any other way?

Reply:
It's probably possible to detect that we're compiling from the command-line. (If that's we're fixing this).